redical_redis/src/commands/rdcl_evt_prune.rs

```diff
@@ -0,0 +1,159 @@
+    let from_timestamp  = timestamp_from_date_string(&from)?;
+    let until_timestamp = timestamp_from_date_string(&until)?;
+
+    if from_timestamp > until_timestamp {
+        return Err(RedisError::String(
+            format!("FROM date: {} cannot be greater than the UNTIL date: {}", &from, &until))
+        );
+    }
+
+    Ok((from_timestamp, until_timestamp))
+}
+
+fn prune_and_reindex(calendar: &mut Calendar, from: i64, until: i64) -> Result<HashMap<String, Box<Event>>, RedisError> {
+    let pruned_events = calendar.prune_events(from, until).unwrap();
+
+    if calendar.indexes_active {
+        for (event_uid, pruned_event) in pruned_events.iter() {
+            let mut calendar_index_updater = CalendarIndexUpdater::new(&event_uid, calendar);
+
+            let updated_event_categories_diff = InvertedEventIndex::diff_indexed_terms(
+                pruned_event.indexed_categories.as_ref(),
+                None,
+            );
+
+            let updated_event_location_type_diff = InvertedEventIndex::diff_indexed_terms(
+                pruned_event.indexed_location_type.as_ref(),
+                None,
+            );
+
+            let updated_event_related_to_diff = InvertedEventIndex::diff_indexed_terms(
+                pruned_event.indexed_related_to.as_ref(),
+                None,
+            );
+
+            let updated_event_geo_diff = InvertedEventIndex::diff_indexed_terms(
+                pruned_event.indexed_geo.as_ref(),
+                None,
+            );
+
+            let updated_event_class_diff = InvertedEventIndex::diff_indexed_terms(
+                pruned_event.indexed_class.as_ref(),
+                None,
+            );
+
+            calendar_index_updater
+                .update_indexed_categories(&updated_event_categories_diff)
+                .map_err(|error| RedisError::String(error.to_string()))?;
+            
+            calendar_index_updater
+                .update_indexed_location_type(&updated_event_location_type_diff)
+                .map_err(|error| RedisError::String(error.to_string()))?;
+            
+            calendar_index_updater
+                .update_indexed_related_to(&updated_event_related_to_diff)
+                .map_err(|error| RedisError::String(error.to_string()))?;
+            
+            calendar_index_updater
+                .update_indexed_geo(&updated_event_geo_diff)
+                .map_err(|error| RedisError::String(error.to_string()))?;
+            
+            calendar_index_updater
+                .update_indexed_class(&updated_event_class_diff)
+                .map_err(|error| RedisError::String(error.to_string()))?;
+        }
+    }
+
+    Ok(pruned_events)
+}
+
+fn notify_keyspace_event(ctx: &Context, calendar_uid: &RedisString, from: &String, until: &String, event_uid: &String) -> Result<(), RedisError> {
+    let event_message = format!("rdcl.evt_prune:{calendar_uid}:{from}-{until}:{event_uid}");
```

Comment:
We should make this broadly consistent with the `evo_prune` keyspace event message (`rdcl.evo_prune:{event_uid}:{occurrence_timestamp}`) 🤔 The calendar uid is included in the keyspace event notification api already which is accessible when receiving the event, so we can omit that 🙂 

I reckon this is good!
```suggestion
    let event_message = format!("rdcl.evt_prune:{event_uid}:{from}-{until}");
```